core/drivers/bcm_hwrng.c

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2019 Broadcom.
+ */
+
+#include <drivers/bcm_hwrng.h>
+#include <initcall.h>
+#include <io.h>
+#include <mm/core_memprot.h>
+#include <mm/core_mmu.h>
+#include <platform_config.h>
+#include <stdlib.h>
+#include <string.h>
+#include <trace.h>
+
+/* Registers */
+#define RNG_CTRL_OFFSET         0x00
+#define RNG_CTRL_MASK           0x00001fff
+#define RNG_CTRL_DISABLE        0x00000000
+#define RNG_CTRL_ENABLE         0x00000001
+
+#define RNG_SOFT_RESET_OFFSET   0x04
+#define RNG_SOFT_RESET_MASK     0x00000001
+
+#define RNG_FIFO_DATA_OFFSET    0x20
+
+#define RNG_FIFO_COUNT_OFFSET   0x24
+
+#define RNG_FIFO_COUNT_MASK     0x000000ff
+#define MAX_WAIT_COUNT          1000
+
+struct bcm_hwrng_dev *bcm_hwrng;
```

Comment:
Instead of lines 32-36 I'd just use a `static vaddr_t hwrng_base;`
It simplifies the code and avoids a call to `malloc()`.